torch/csrc/api/src/nn/modules/adaptive.cpp

```diff
@@ -53,6 +53,10 @@ void AdaptiveLogSoftmaxWithLossImpl::reset() {
   tail = this->register_module("tail", ModuleList());
 
   for (const auto i : c10::irange(n_clusters)) {
+    TORCH_CHECK(
+        static_cast<int64_t>(std::pow(options.div_value(), (i + 1))) != 0,
+        "the intermediate value calculated from div_value should not be equal to 0");
+
     int64_t hsz = options.in_features() /
```

Comment:
Looking at this again, it seems like there is an inconsistency between the Python and C++ API here

In this case, 

```cpp
int64_t hsz = options.in_features() / static_cast<int64_t>(std::pow(options.div_value(), (i + 1)));
```
will be 
```
int64_t hsz = 16 / static_cast<int64_t>(std::pow(options.div_value(), (i + 1)));
```

which gives 16/0

on the other hand in the python API the corresponding line 

https://github.com/pytorch/pytorch/blob/73a661abf1f0d724dd71a8b3b0da53eb5c8524a0/torch/nn/modules/adaptive.py#L158

would give 

```
int(16//0.25 ** 1) = 64
```

I did not notice that the issue used 0 for the `div_value` in Python rather than 0.25 (which was used for C++), which does not error

```python
import torch
m = torch.nn.AdaptiveLogSoftmaxWithLoss(1,3,(1,),div_value=0.25)
```

Instead of a check on the denominator, maybe we need to do 
```cpp
static_cast<int64_t>(std::floor(options.in_features() / std::pow(options.div_value(), (i + 1))));
```

And also a separate `TORCH_CHECK` that `div_value` is not 0 (iiuc that is the only case where std::pow(div_value, i) will be 0)

Apologies for the churn! Let me know if you agree with the above analysis

Reply:
Sorry, It was my mistake, I agree with you very much, and have fixed it again, please have a look at it again.
Thank you.

`static_cast<int64_t>` can round down also.